src/baal/utils/metrics.py

```diff
@@ -307,3 +326,73 @@ def value(self):
         precision = tp / np.maximum(1, tp + fp)
         recall = tp / np.maximum(1, tp + fn)
         return {'accuracy': acc, 'precision': precision, 'recall': recall}
+
+
+@dataclasses.dataclass
+class Report:
+    tp: int = 0
+    fp: int = 0
+    fn: int = 0
+
+
+class PRAuC(Metrics):
+    """
+    Precision-Recall Area under the curve.
+
+    Args:
+        num_classes (int): Number of classes
+        n_bins (int): number of confidence threshold to evaluate on.
+        average (bool): If true will return the mean AuC of all classes.
+    """
+
+    def __init__(self, num_classes, n_bins, average):
+        self.num_classes = num_classes
+        self.threshold = np.linspace(0.02, 0.99, n_bins)
+        self._data = defaultdict(lambda: defaultdict(lambda: Report()))
+        super().__init__(average)
+
+    def reset(self):
+        self._data.clear()
+
+    def update(self, output=None, target=None):
+        """
+        Update the confusion matrice according to output and target.
+
+        Args:
+            output (tensor): predictions of model
+            target (tensor): labels
+        """
+        output = output.detach().cpu().numpy()
+        target = target.detach().cpu().numpy()
+        output = to_prob(output)
```

Comment:
this is kind of risky if the output is already softmaxed. Wouldn't it be better if you do an assertion instead? For instance, you make this function expect probabilities and and assert that the probs add up to 1. This reduces the amount possible silent bugs.

Reply:
If the output sums to 1 and they are between 0 and 1 we don't softmax it.

Reply:
ah oki, makes sense!